src/skspatial/objects/cylinder.py

```diff
@@ -597,3 +692,106 @@ def _intersect_line_with_finite_cylinder(
+    initial_direction = Line.best_fit(points).vector.unit()
+    return cartesian_to_spherical(*initial_direction)
+
+
+def _compute_projection_matrix(direction: Vector) -> np.ndarray:
+    """Compute the projection matrix."""
+    return np.identity(3) - np.dot(np.reshape(direction, (3, 1)), np.reshape(direction, (1, 3)))
+
+
+def _compute_skew_matrix(direction: Vector) -> np.ndarray:
+    """Compute the skew matrix."""
+    return np.array(
+        [
+            [0.0, -direction[2], direction[1]],
+            [direction[2], 0.0, -direction[0]],
+            [-direction[1], direction[0], 0.0],
+        ],
+    )
+
+
+def _compute_a_matrix(input_samples: List[np.ndarray]) -> np.ndarray:
+    """Compute the :math:`{A}` matrix."""
+    return sum(np.dot(np.reshape(sample, (3, 1)), np.reshape(sample, (1, 3))) for sample in input_samples)
+
+
+def _compute_a_hat_matrix(a_matrix: np.ndarray, skew_matrix: np.ndarray) -> np.ndarray:
+    r"""Compute the :math:`\\hat{A}` matrix."""
+    return np.dot(skew_matrix, np.dot(a_matrix, np.transpose(skew_matrix)))
+
+
+def _compute_g(direction: Vector, points: Points) -> float:
+    """Compute :math:`G`."""
+    projection_matrix = _compute_projection_matrix(direction)
+    skew_matrix = _compute_skew_matrix(direction)
+    input_samples = [np.dot(projection_matrix, X) for X in points]
+    a_matrix = _compute_a_matrix(input_samples)
+    a_hat_matrix = _compute_a_hat_matrix(a_matrix, skew_matrix)
+
+    u = sum(np.dot(sample, sample) for sample in input_samples) / len(points)
+    v = np.dot(a_hat_matrix, sum(np.dot(sample, sample) * sample for sample in input_samples)) / np.trace(
+        np.dot(a_hat_matrix, a_matrix),
+    )
+    return sum((np.dot(sample, sample) - u - 2 * np.dot(sample, v)) ** 2 for sample in input_samples)
+
+
+def _compute_center(direction: Vector, points: Points) -> Point:
+    """Compute center."""
+    projection_matrix = _compute_projection_matrix(direction)
+    skew_matrix = _compute_skew_matrix(direction)
+    input_samples = [np.dot(projection_matrix, X) for X in points]
+    a_matrix = _compute_a_matrix(input_samples)
+    a_hat_matrix = _compute_a_hat_matrix(a_matrix, skew_matrix)
+
+    return np.dot(a_hat_matrix, sum(np.dot(sample, sample) * sample for sample in input_samples)) / np.trace(
+        np.dot(a_hat_matrix, a_matrix),
+    )
+
+
+def _compute_radius(direction: Vector, points) -> float:
+    """Compute radius."""
+    projection_matrix = _compute_projection_matrix(direction)
+    center = _compute_center(direction, points)
+    return np.sqrt(
+        sum(np.dot(center - point, np.dot(projection_matrix, center - point)) for point in points) / len(points),
+    )
+
+
+def cartesian_to_spherical(x: float, y: float, z: float) -> _SphericalCoordinates:
+    """Convert cartesian to spherical coordinates."""
+    theta = np.arccos(z / np.sqrt(x**2 + y**2 + z**2))
+
+    if math.isclose(x, 0.0, abs_tol=1e-9) and math.isclose(y, 0.0, abs_tol=1e-9):
```

Comment:
Since I am comparing to 0, it should be a good practice to define an absolute tolerance. I have not modified this line for the time being. Let me know your thoughts about it, thanks!

Reply:
`math.isclose` already has a default value for `abs_tol`. If the test passes using that default, I'd rather just use it to make the code simpler.

Reply:
Yes, you are right! `abs_tol` has a default value, unfortunately is `0.0` so:
```python
>>> math.isclose(0.0, 1e-31)
False
>>> math.isclose(0.0, 1e-31, abs_tol=1e-9)
True
```
Where I am not comparing to zero I have already removed the `abs_tol`. If you find `abs_tol=1e-9` in the unit tests, then it means it's necessary for the tests to pass.
I hope that I have properly understood your question. Thanks.